src/macaron/repo_finder/commit_finder.py

```diff
@@ -40,49 +40,27 @@ def get_commit_from_version(git_obj: Git, purl: PackageURL) -> tuple[str, str]:
         logger.debug("Missing version for artifact: %s", purl.name)
         return "", ""
     logger.debug("Searching for commit of artifact version using tags: %s@%s", purl.name, purl.version)
-    # Create version based regex from the PackageURL for matching against tags.
-    # The version is split on non-alphanumeric characters to separate the version parts from the non-version parts.
-    # e.g. 1.2.3-DEV -> [1, 2, 3, DEV]
-    split = split_pattern.split(purl.version)
-    logger.debug("Split version: %s", split)
-    if not split:
-        split = [purl.version]
-    this_version_pattern = ""
-    for part in split:
-        # Validate the split part by checking it is only comprised of alphanumeric characters.
-        valid = validation_pattern.match(part)
-        if not valid:
-            continue
-        if this_version_pattern:
-            # To maximise chances of matching tags, the regex between the version parts allows for zero or more
-            # non-numeric characters.
-            this_version_pattern = this_version_pattern + INFIX
-        this_version_pattern = this_version_pattern + str(part)
-    # Prepend the optional prefix, add a named capture group for the version, and enforce end of string analysis.
-    this_version_pattern = PREFIX + "(?P<version>" + this_version_pattern + ")" + "$"
-    logger.debug("Created pattern: %s", this_version_pattern)
-    target_version_pattern = re.compile(this_version_pattern, flags=re.IGNORECASE)
 
-    # A pattern for very weakly confirming the presence of a version.
-    has_version_pattern = re.compile(".*[0-9].*", flags=re.IGNORECASE)
+    target_version_pattern = _build_version_pattern(purl.version)
+    has_name_pattern = re.compile(f".*{purl.name}.*[0-9].*", flags=re.IGNORECASE)
```

Comment:
Would there be any case where there would be non-alphanumeric characters in `purl.name` that could change the semantic of this regex string? If yes, I think we should escape those or use a [raw string](https://docs.python.org/3/library/re.html#raw-string-notation) instead.